### ISSUE TITLE
[P1][dogfood] nav action=impact / --codegraph-impact: risk.level="critical" (score 100) maps to verdict=INFO + "Low risk change — proceed with edit" (missing "critical" branch in _impact_verdict inverts the edit-safety decision surface)

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#798]
**Severity:** P1  |  **Surface:** MCP nav action=impact (mode=function_impact) and CLI --codegraph-impact --codegraph-impact-mode function_impact; root cause tree_sitter_analyzer/mcp/tools/codegraph_impact_tool.py:_impact_verdict (lines 583-594), with the next_step branch at lines 488-496 and the 4-level emitter at lines 180-187.
_Found by parallel dogfood audit round 3 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked._

### Observed
Both CLI and MCP return risk.score=100, risk.level="critical", fan_in=114, cross_file_callers=64, but verdict=INFO and agent_summary.next_step="Low risk change — proceed with edit; run nearest test file afterwards." The summary_line is internally contradictory: "impact: 'apply_toon_format_to_response' risk=critical verdict=INFO". _impact_verdict has no branch for level=="critical" so it falls through `return "INFO"`; the next_step branch then treats INFO as the low-risk/proceed path. Net: every function in the highest-risk bucket (score>=60, biggest blast radius) is reported to agents as safe to edit. The medium-risk format_as_toon correctly yields verdict=REVIEW, confirming the two are inverted relative to actual risk.

### Expected
The verdict and next_step must be monotonic with risk.level: the most dangerous category (critical) must get the strongest warning, never the weakest. critical should map to a blocking/caution verdict (e.g. CAUTION or a dedicated DANGER) and next_step must warn ("trace callers and run tests before editing"), never "Low risk change — proceed with edit". Add a `if level == "critical": return "CAUTION"` (or stronger) branch to _impact_verdict (codegraph_impact_tool.py ~line 588) so the decision surface is monotonic with risk.score. A RED-first test should assert level=critical -> verdict in {CAUTION, DANGER} and that next_step does not contain "proceed with edit".

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood
# CLI:
uv run python -m tree_sitter_analyzer --codegraph-impact apply_toon_format_to_response --codegraph-impact-mode function_impact --output-format json 2>/dev/null | python3 -c "import json,sys; d=json.load(sys.stdin); print('verdict:',d['verdict'],'| risk.level:',d['risk']['level'],'| risk.score:',d['risk']['score'],'| fan_in:',d['risk']['factors']['fan_in']); print('next_step:',d['agent_summary']['next_step'])"
# Expect (BUG): verdict: INFO | risk.level: critical | risk.score: 100 | fan_in: 114 ; next_step: Low risk change — proceed with edit...
# MCP: nav action=impact symbol=apply_toon_format_to_response output_format=json -> same (verdict=INFO, risk.level=critical).
# Inverse (correct) control: symbol=format_as_toon -> risk.level=medium, verdict=REVIEW, next_step "High risk change — trace callers...".
```

### Verifier notes
Re-reproduced on BOTH surfaces against develop @ 471ac5f5. MCP nav action=impact symbol=apply_toon_format_to_response (default mode=function_impact) and CLI --codegraph-impact ... --codegraph-impact-mode function_impact --output-format json both return risk.score=100, risk.level="critical", fan_in=114, yet verdict=INFO with agent_summary.next_step="Low risk change — proceed with edit; run nearest test file afterwards.". The summary_line itself is self-contradictory: "impact: 'apply_toon_format_to_response' risk=critical verdict=INFO". Inverse case confirms the inversion: format_as_toon (risk.score=25, level=medium) correctly yields verdict=REVIEW + "High risk change — trace callers... before editing". So the higher-risk function gets the weaker verdict and the lower-risk function gets the stronger one. Root cause verified in source: _compute_risk_score (codegraph_impact_tool.py:180-187) emits 4 levels (critical>=60, high>=40, medium>=20, low<20), but _impact_verdict (lines 583-594) only maps unknown->NOT_FOUND, high->CAUTION, medium->REVIEW, low->INFO and falls through to `return "INFO"` for any other level — there is NO "critical" branch, so critical falls to INFO. Then next_step (lines 488-496) treats INFO as the low-risk/proceed path. Deterministic across CLI and MCP runs. Not a flag/user error: both invocations use documented flags/actions exactly as prescribed; mode defaults to function_impact. Dup check: #782 is a DIFFERENT bug (--change-impact tool; top-level verdict vs agent_summary.verdict disagree WITHIN one response = two independent verdict computations) and #781 is --safe-to-edit summary_line-vs-structured mismatch; neither covers the codegraph_impact critical->INFO mapping gap where verdict+next_step are mutually consistent but both inverted vs risk.level. No open issue covers this. P1 justified: it produces a false-green "safe to edit" on exactly the highest-blast-radius functions, mirroring P1 findings #754/#780.

**Comment by aimasteracc** [comment]
Fixed and merged to develop. See commit in the linked PR.
